subprojects/gradle/instrumentation-tests/src/integTest/kotlin/com/avito/instrumentation/reservation/client/kubernetes/KubernetesReservationClientIntegrationTest.kt

```diff
@@ -0,0 +1,173 @@
+package com.avito.instrumentation.reservation.client.kubernetes
+
+import com.avito.instrumentation.reservation.adb.AndroidDebugBridge
+import com.avito.instrumentation.reservation.adb.EmulatorsLogsReporter
+import com.avito.instrumentation.reservation.client.ReservationClient
+import com.avito.instrumentation.reservation.request.Device.CloudEmulator
+import com.avito.instrumentation.reservation.request.Reservation
+import com.avito.instrumentation.reservation.request.createStubInstance
+import com.avito.runner.service.worker.device.adb.Adb
+import com.avito.truth.assertThat
+import com.avito.utils.gradle.KubernetesCredentials
+import com.avito.utils.gradle.createKubernetesClient
+import com.avito.utils.logging.FakeCILogger
+import com.google.common.truth.Truth.assertThat
+import io.fabric8.kubernetes.client.KubernetesClientException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import java.io.File
+import java.net.UnknownHostException
+
+internal class KubernetesReservationClientIntegrationTest {
+
+    private val logger = FakeCILogger()
+    private val deploymentNameGenerator = FakeDeploymentNameGenerator()
+    private var clientOne: ReservationClient? = null
+    private var clientTwo: ReservationClient? = null
+
+    @Test
+    fun `claim - throws exception - unknown host`() {
+        clientOne = createClient(
+            kubernetesUrl = "unknown-host",
+            kubernetesNamespace = "emulators"
+        )
+
+        val exception = assertThrows<KubernetesClientException> {
+            runBlocking {
+                clientOne!!.claim(
```

Comment:
Why do we need to force unwrap?

Reply:
because it's nullable type; couldn't use lateinit here